internal/verifier/verifier_test.go

```diff
@@ -34,11 +37,47 @@ func TestReproducibleProvenanceVerifier(t *testing.T) {
 	testutil.Chdir(t, "../../")
 	verifier := ReproducibleProvenanceVerifier{}
 
-	if err := verifier.Verify(schemaExamplePath); err != nil {
+	if err := verifier.Verify(validProvenancePath); err != nil {
 		t.Fatalf("couldn't verify the provenance file: %v", err)
 	}
 }
 
+func TestReproducibleProvenanceVerifier_invalidHash(t *testing.T) {
+	// The path to provenance is specified relative to the root of the repo, so we need to go one level up.
+	// Get the current directory before that to restore the path at the end of the test.
```

Comment:
Why does the path need to be restored? Does it mean that it affects the execution of other test cases? That would seem problematic, since the execution of the tests would differ whether they are running concurrently or not. Is it possible to make the entire process not depend on the PWD and instead use absolute paths for everything? It would still be possible here to concat PWD with a relative path to obtain an absolute path and use that, but at least that would not mutate global state.

Reply:
This is basically because when provenances are loaded, they are also validated against the schema. The path of the schema is hardcoded inside some other file, and getting the path right requires changing the directory. 

I have not been able to come up with a reliable and neat solution yet, but it certainly needs to be fixed in a separate PR. We have this bit of code in all tests, and it is causing a lot of clutter.

We plan to switch to the SLSA format, and retire the amber provenance format. So it probably is better to wait for that change. I believe this problem will automatically go away when that change is completed. Created #127.

Reply:
+1 I think the basic functions (which are tested here, e.g. `verifier.Verify`) should deal with the actual content of the files (as strings), they should not assume they can read a file by its path. Then we can have helpers to just read the file at a path, and we can compose that by passing its content to the function in question.